messages/teleporter/message_handler.go

```diff
@@ -236,6 +240,37 @@ func (m *messageHandler) SendMessage(signedMessage *warp.Message, destinationCli
 		)
 		return err
 	}
+
+	// Wait for the message to be included in a block before returning
+	callCtx, callCtxCancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer callCtxCancel()
+	receipt, err := utils.CallWithRetry[*types.Receipt](
+		callCtx,
+		func() (*types.Receipt, error) {
+			return destinationClient.Client().(ethclient.Client).TransactionReceipt(callCtx, txHash)
+		},
+	)
+	if err != nil {
+		m.logger.Error(
+			"Failed to get transaction receipt",
+			zap.String("destinationBlockchainID", destinationBlockchainID.String()),
+			zap.String("warpMessageID", signedMessage.ID().String()),
+			zap.String("teleporterMessageID", teleporterMessageID.String()),
+			zap.Error(err),
+		)
+		return err
+	}
+	if receipt.Status != types.ReceiptStatusSuccessful {
+		m.logger.Error(
+			"Transaction failed",
+			zap.String("destinationBlockchainID", destinationBlockchainID.String()),
+			zap.String("warpMessageID", signedMessage.ID().String()),
+			zap.String("teleporterMessageID", teleporterMessageID.String()),
+			zap.String("txHash", txHash.String()),
+		)
+		return fmt.Errorf("transaction failed with status: %d", receipt.Status)
+	}
```

Comment:
Can we move this section out to a helper? It also is close to duplicating the utility functions we have in Teleporter for use in the E2E such as [WaitMined](https://github.com/ava-labs/teleporter/blob/main/tests/utils/utils.go#L584). Would it make sense to export those from a non-test package to use here?

Reply:
I think this logic should continue to live here for a couple reasons:
- Moving `WaitMined` to a non-test utility in `teleporter` doesn't make a ton of sense, since there's nothing specific to Teleporter in that function. It's really a utility to support our current test framework
- `awm-relayer` has slightly different semantics. In `WaitMined`, we also wait for receipt's block height to account for differences in network views among nodes behind the public API. This is necessary to avoid races in the tests when run on Fuji. In the relayer, we instead assume that a valid receipt indicates that the tx will eventually be seen by all nodes.